src/chapters/errors-and-debugging/diagnosing-error-messages.rst

```diff
@@ -0,0 +1,150 @@
+               ^^^^^^^^^^^^^^
+
+.. index:: ! token
+
+JavaScript is telling us that in the area of ``"Hello, name);`` it encountered an invalid token. **Token** is a fancy word that means a symbol, variable, or other atomic element of a program. In this case, the invalid token is ``"Hello, name);``. JavaScript sees the double-quote character and expects a string. However, the string does not have a closing ``"``, making it invalid.
+
+Fixing this error gives us the program:
+
+.. sourcecode:: js
+
+   let name = Julie;
+   console.log("Hello", name);
+
+
+Syntax Errors and Code Highlighting
+-----------------------------------
+
+.. index::
+   single: syntax; highlighting
+
+Most editors in which you are likely to write code will provide a feature known as **syntax highlighting**. Such code editors will highlight different types of tokens in different ways. For example, strings may be red, while variables may be green. This useful feature can help you quickly identify syntax errors visually. 
+
+As an example, let's look at a screenshot of the code we have been working with in this section, from within an `editor at repl.it <https://repl.it/@launchcode/Syntax-Highlighting>`_.
+
+.. figure:: figures/syntax-highlighting.png
+   
+   Screenshot of a program with two syntax errors
+
+Notice that the string ``Hello`` is colored red, while *most* of the symbols (``=``, ``;``, ``.``, and ``(``) are colored black. At the end of line 1, however, the final ``)`` and ``;`` are both red rather than black. Since we haven't closed the string, the editor assumes that these two symbols are *part of* the string. Since we expect such symbols to be black in this editor, the difference in color is a clue that something is wrong with our syntax.
+
+
+A Runtime Error
+---------------
+
+Having fixed the syntax error, we can now run our program again. Doing so displays yet another error.
+
+::
+
+   Hello
+   /Users/chris/dev/sandbox/js/syntax.js:1
+   let name = Julie;
+            ^
+
+   ReferenceError: Julie is not defined
+      at Object.<anonymous> (/Users/chris/dev/sandbox/js/syntax.js:1:74)
+      at Module._compile (internal/modules/cjs/loader.js:738:30)
+      at Object.Module._extensions..js (internal/modules/cjs/loader.js:749:10)
+      at Module.load (internal/modules/cjs/loader.js:630:32)
+      at tryModuleLoad (internal/modules/cjs/loader.js:570:12)
+      at Function.Module._load (internal/modules/cjs/loader.js:562:3)
+      at Function.Module.runMain (internal/modules/cjs/loader.js:801:12)
+      at internal/main/run_main_module.js:21:11
+
+
+We have a new error message, this time having to due with line 1 of our code. We didn't see this error before because it is a runtime error, and the program never reached the execution phase earlier. Due to the syntax error on line 2, it stopped during the parsing phase. Even though the current error appears *before* the original syntax error in the code, the syntax error was reported first.
```

Comment:
"...this time involving line 1 of our code."

We can remove "and the program never reached the execution phase earlier" with no loss of meaning.

Replace "it stopped" with "the program stopped".

"Even though the runtime error involves the line *before* the original syntax error, the syntax error still gets reported first."